proposed/swift-interop.md

```diff
@@ -152,33 +152,40 @@ If possible, Swift tuples should be represented as `ValueTuple`s in .NET. If thi
 
 #### Projection Tooling Components
 
-The projection tooling should be split into these components:
+The projection tooling will be based on the [Binding Tools for Swift](https://github.com/xamarin/binding-tools-for-swift) (BTfS). The BTfS contains a set of tools that can consume a compiled Apple Swift library and generate wrappers (bindings) that allow it to be surfaced as a .NET library. This tool will be implemented as a self-hosted .NET CLI tool, automating the generation of wrappers by parsing the Swift library interface. It generates C# bindings that can be utilized as a .NET library and Swift bindings that C# bindings invoke in cases where direct P/Invoke into Swift library is not possible.
+
+The projection tooling should be split into these components.
 
 ##### Importing Swift into .NET
 
-1. A tool that takes in a `.swiftinterface` file or Swift sources and produces C# code.
-2. A library that provides the basic support for Swift interop that the generated code builds on.
-3. User tooling to easily generate Swift projections for a given set of `.framework`s.
-    - This tooling would build a higher-level interface on top of the tool in item 1 that is more user-friendly and project-system-integrated.
-4. (optional) A NuGet package, possibly referencable by `FrameworkReference` or automatically included when targeting macOS, Mac Catalyst, iOS, or tvOS platforms that exposes the platform APIs for each `.framework` that is exposed from Swift to .NET.
-    - This would be required to provide a single source of truth for Swift types so they can be exposed across an assembly boundary.
+Importing Swift into .NET is done through the following steps:
+
+1. The tool analyzes a source `.swift` library and maps entry points to mangled names.
+2. Then,the tool aggregates the public API; it takes in a `.swiftmodule` file (similar to a C header file) generated by the Swift compiler, creating a module declaration.
```

Comment:
So the tool takes two files: `.swift` library and matching `.swiftmodule` file as the input. Do I understand it correctly?

Can we generate everything from `.swiftmodule` to simplify the flow?

Reply:
The tool takes `.swiftmodule` and internal mapping (entry points to mangled names) from step 1 and create a module declaration. I assume it is possible to create the internal mapping using `.swiftmodule` instead of a source `.swift` library.

Reply:
> create the internal mapping using .swiftmodule instead of a source .swift library.

I think that would be better. Simplifies the flow and avoids a whole class of problems where the two input files do not match.

Reply:
Created a tracking issue for improvements: https://github.com/xamarin/binding-tools-for-swift/issues/806

Reply:
Can we take only a .swift library? In theory we can, but we probably don't want to. The main reason for having the swift library is for getting at the entry points and the type metadata which is done by demangling the symbols into a manipulatable type. This is a straightforward process that, today, is done via a direct port of Apple's demangler. The different challenge with writing the mangler is that unless things have changed substantially, it's not centralized code that we could port easily - but it's certainly worth looking at as a possibility.